model/articles.model.js

```diff
@@ -38,3 +38,42 @@ exports.checkArticleExist = (article_id)=>{
     })
 }
 
+exports.updateVote = (article_id, inc_votes) => {
+    if (inc_votes === 0 || typeof inc_votes !== 'number') {
+      return Promise.reject({ status: 400, msg: 'Bad request' });
+    }
+  
+    return db.query('SELECT votes FROM articles WHERE article_id = $1', [article_id])
+      .then((result) => {
+        if (result.rows.length === 0) {
+          return Promise.reject({ status: 404, msg: 'Article not found' });
+        }
+  
+        const currentVotes = result.rows[0].votes;
+        const updatedVotes = currentVotes + inc_votes;
+  
+        if (updatedVotes < 0) {
+          return Promise.reject({ status: 400, msg: 'Bad request' });
+        }
+  
+        let updateQuery;
+        let queryParams;
+  
+        if (inc_votes > 0) {
+          updateQuery = `UPDATE articles SET votes = votes + $1 WHERE article_id = $2 RETURNING *`;
+          queryParams = [inc_votes, article_id];
+        } else {
+          updateQuery = `UPDATE articles SET votes = votes - $1 WHERE article_id = $2 RETURNING *`;
+          queryParams = [-inc_votes, article_id];
+        }
```

Comment:
You dont have to do this check :)
Just adding is fine!
Take a look at the following:
10 + 1 ===11
10 + (-1) === 9

When you add a negative number, it still takes away, so you'll only need the version of this code on line 63.